nrf_security/src/psa_crypto_driver_wrappers.c

```diff
@@ -97,6 +98,216 @@
 #include "psa_crypto_se.h"
 #endif
 
+
+/**
+ * @brief This can be used to extract just the base value of algorithms that
+ *  are combined with a hashing scheme. 
+ * @param alg The full algorithm identifier.
+ */
+#define GET_ALG_BASE(alg) (alg & (~PSA_ALG_HASH_MASK))
+
+/**
+ * @brief Compare AEAD alogrithms ingoring the the output tag length
+ * 
+ */
+#define EQUAL_AEAD_ALG(alg_1, alg_2) ((alg_1 & ~PSA_ALG_AEAD_TAG_LENGTH_MASK) \
+                                == (alg_2 & ~PSA_ALG_AEAD_TAG_LENGTH_MASK)) 
+
+/**
+ * @brief Checks if the given algorithm is one the following ECDSA algorithms
+ * PSA_ALG_DETERMINISTIC_ECDSA
+ * PSA_ALG_ECDSA
+ * PSA_ALG_ECDSA_ANY PSA_ALG_ECDH
+ * or the PSA_ALG_ECDH 
+ */
+static bool is_ecdsa_or_ecdh(psa_algorithm_t algorithm){
+    return (GET_ALG_BASE(algorithm) == PSA_ALG_DETERMINISTIC_ECDSA_BASE  ||
```

Comment:
I think that it is better to use the defined macros from PSA.
Look at: PSA_ALG_ECDSA_IS_DETERMINISTIC(alg)
PSA_ALG_IS_ECDSA(alg)
PSA_ALG_IS_ECDH(alg)

They are basically doing the same thing.